assets/js/src/fade.js

```diff
@@ -1,35 +1,59 @@
-const mobileOffset = 270;
-// Start with nav-work button highlighted and only work section showing
-document.getElementById('nav-work').classList.add('current-btn');
-document.getElementById('work').classList.add('display');
+const headerElement = document.getElementById('header');
+const headerGeometry = headerElement.getBoundingClientRect();
+const stickyNavOffset = headerGeometry.height + window.scrollY + headerGeometry.y;
+const navElements = Array.from(document.getElementsByClassName('nav-btn'));
+const sectionElements = Array.from(document.getElementsByTagName('section'));
+const fadeElement = document.getElementById('fade-wrapper');
 
-const navElements = document.getElementsByClassName('nav-btn');
+function selectWithLocation(el, locationHash) {
+  return el.dataset.section === locationHash.substring(1);
+}
 
-Array.from(navElements).forEach((navElement) => {
-  navElement.addEventListener('click', (e) => {
-    e.preventDefault();
-    Array.from(document.getElementsByTagName('li'))
-      .forEach(el => el.classList.remove('current-btn'));
-    document.getElementById(e.target.id).classList.add('current-btn');
-    $('#fade-wrapper').fadeOut(500, () => {
-      Array.from(document.getElementsByTagName('section'))
-        .forEach(el => el.classList.remove('display'));
-      document.getElementById(e.target.id.slice(4)).classList.add('display');
-      if (window.scrollY > mobileOffset) {
-        window.scrollTo(0, mobileOffset);
-      }
-      $('#fade-wrapper').fadeIn(500);
-    });
+function getActiveElements(locationHash) {
+  if (locationHash && locationHash !== '') {
+    const navElement = navElements.find(el => selectWithLocation(el, locationHash));
+    const sectionElement = sectionElements.find(el => selectWithLocation(el, locationHash));
+    return [navElement, sectionElement];
+  }
+  return [navElements[0], sectionElements[0]];
+}
+
+// Page load could be index.html, or index.html#contact etc.
+const activeElements = getActiveElements(window.location.hash);
+activeElements[0].classList.add('current-btn');
+activeElements[1].classList.add('display');
+
+window.addEventListener('hashchange', (e) => {
+  e.preventDefault();
+  navElements.forEach((el) => {
+    el.classList.remove('current-btn');
   });
+  const elements = getActiveElements(window.location.hash);
```

Comment:
You could use array destructuring here:

```suggestion
  const [navElement, sectionElement] = getActiveElements(window.location.hash);
```

It's a nice way to handle functions which return a tuple.

If you don't go with the destrcuturing approach, you should at least give the two values variable names. Later in the code, it's very hard to reason about what `elements[1]` is.

That said, see above for my suggestion which avoids returning a tuple all together.

Reply:
Makes sense